bin/check-hardware-fail.rb

```diff
@@ -29,9 +29,26 @@
 require 'sensu-plugin/check/cli'
 
 class CheckHardwareFail < Sensu::Plugin::Check::CLI
+  option :lines,
+         short: '-l NUMBER',
+         long: '--lines NUMBER',
+         proc: proc(&:to_i),
+         default: 0,
+         description: 'Maximum number of lines to read from dmesg, 0 (default) means all'
+
+  option :query,
+         short: '-q QUERY',
+         long: '--query QUERY',
+         default: 'Hardware Error',
+         description: 'What to look for in the output of dmesg'
+
   def run
-    errors = `dmesg`.lines.select { |l| l['[Hardware Error]'] }
-    critical 'Hardware Error Detected' if errors.any?
-    ok 'Hardware OK'
+    errors = if config[:lines] == 0
+               `dmesg`.lines.select { |l| l[/#{config[:query]}/] }
+             else
+               `dmesg | tail -n #{config[:lines]}`.lines.select { |l| l[/#{config[:query]}/] }
```

Comment:
I thought about what you said regarding using `head` and I've added an `--invert` option, that will make the plugin use `head` instead of `tail`.
I've also added a line to check if the system command was successfully executed and return with unknown if the command failed.